ui/imports/shared/controls/CurrencyAmountInput.qml

```diff
@@ -0,0 +1,95 @@
+﻿import QtQuick 2.15
+import QtQuick.Controls 2.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core.Theme 0.1
+
+import utils 1.0
+
+TextField {
+    id: root
+
+    property alias value: internalProp.value // accepts double/float or string representation, rejects NaN
+    readonly property bool valid: acceptableInput
+
+    property int decimals: 2 // number of decimal places to display
+    property string currencySymbol: "USD" // currency symbol, optional
+    property double minValue: Number.MIN_VALUE // min value
+    property double maxValue: Number.MAX_VALUE // max value
+    readonly property string valueAsString: internalProp.valueAsString
+
+    FormattedDoubleProperty {
+        id: internalProp
+        onValueChanged: root.text = asString() // min number of decimals, strip zeroes
+    }
+
+    Keys.onPressed: (event) => {
+                        // additionally accept dot (.) and convert it to the correct decimal point char
+                        if (event.key === Qt.Key_Period) {
```

Comment:
Should't `Key_Comma` be handled in the same way? For some locale period is the decimal char.

Reply:
Then that would be the locale's decimal point and we don't need to do anything there